src/Data/Text/Builder/Linear/Buffer.hs

```diff
@@ -193,3 +198,61 @@ foldlIntoBuffer f = go
     go ∷ Buffer ⊸ [a] → Buffer
     go !acc [] = acc
     go !acc (x : xs) = go (f acc x) xs
+
+-- $custom_hexadecimal
+--
+-- Note that no /upper/ case hexadecimal formatting is provided. This package
+-- provides a minimal API with utility functions only for common cases. For
+-- other use cases, please adapt the code of this package, e.g. as shown hereinafter.
+--
+-- __Example: Unicode code points__
+--
+-- The following example illustrates how to implement an efficient function that displays
+-- the [Unicode code point](https://en.wikipedia.org/wiki/Unicode#Codespace_and_code_points)
+-- of a character using the normative notation @U+NNNN@, such as @U+005A@ for “Z”. It
+-- involves /upper-case/ and /right-justifying/ with “0”.
+--
+-- >>> :set -XLinearTypes -XMagicHash
+-- >>> import Data.Char (ord)
+-- >>> import Data.Bits (Bits(..), FiniteBits(..))
+-- >>> import qualified Data.Text.Array as A
+-- >>> import Data.Text.Builder.Linear.Core
+-- >>> import GHC.Exts
+-- >>> import GHC.ST (ST(..))
+-- >>> :{
+-- (|>&&) :: Buffer %1 -> Char -> Buffer
+-- (|>&&) buf ch =
+--   appendBounded 8 (\dst dstOff -> unsafeAppendCodePoint dst dstOff (ord ch)) buf
```

Comment:
Why not `appendExact 6`?

Reply:
If that's because code points outside of BMP need up to 8 characters, it would be nice to call out this intricacy in the description of the problem.

Reply:
```haskell
maxBound @Char
-- '\x10ffff'
length "U+10FFFF"
-- 8
```

Reply:
So length would be between 6 (e.g. `U+0000`) and 8 (e.g. `U+10FFFF`).

Reply:
Did not get your 2nd message in time. Yes, I should update the description to make this clear. Again, this is just an example for discussion: the idea is to provide examples of writing efficient formatting functions that would not fit in a minimalistic library.

Reply:
Yeah, I'm fine with adding such piece of documentation, thanks.

Reply:
Done